application-admintools-default/src/main/java/com/xwiki/admintools/internal/InstanceUsage.java

```diff
@@ -0,0 +1,184 @@
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package com.xwiki.admintools.internal;
+
+import java.text.DecimalFormat;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.script.ScriptContext;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.slf4j.Logger;
+import org.xwiki.component.annotation.Component;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryFilter;
+import org.xwiki.query.QueryManager;
+import org.xwiki.script.ScriptContextManager;
+import org.xwiki.template.TemplateManager;
+import org.xwiki.wiki.descriptor.WikiDescriptor;
+import org.xwiki.wiki.descriptor.WikiDescriptorManager;
+
+import com.xwiki.admintools.WikiSizeResult;
+import com.xwiki.admintools.internal.data.identifiers.CurrentServer;
+
+/**
+ * Access info about the size of the existing Wikis.
+ *
+ * @version $Id$
+ */
+@Component(roles = InstanceUsage.class)
+@Singleton
+public class InstanceUsage
+{
+    private static final String TEMPLATE_NAME = "wikiSizeTemplate.vm";
+
+    @Inject
+    private PingProvider pingProvider;
+
+    @Inject
+    private WikiDescriptorManager wikiDescriptorManager;
+
+    @Inject
+    private QueryManager queryManager;
+
+    @Inject
+    @Named("count")
+    private QueryFilter countFilter;
+
+    @Inject
+    private Logger logger;
+
+    @Inject
+    private CurrentServer currentServer;
+
+    @Inject
+    private TemplateManager templateManager;
+
+    @Inject
+    private ScriptContextManager scriptContextManager;
+
+    /**
+     * Get the data in a format given by the associated template.
+     *
+     * @return the rendered template as a {@link String}.
+     */
+    public String renderTemplate()
+    {
+        try {
+            ScriptContext scriptContext = this.scriptContextManager.getScriptContext();
+            boolean found = currentServer.getCurrentServer() != null;
+            scriptContext.setAttribute("found", found, ScriptContext.ENGINE_SCOPE);
+            if (!found) {
+                this.logger.error("Used server not found!");
+                return this.templateManager.render(TEMPLATE_NAME);
+            }
+            List<WikiSizeResult> wikisInfo = getWikisSizeInfo();
```

Comment:
I feel that this component is doing too much, so I'll move the code that collects this wikiSizeInfo in a different component. WDYT about (namings are up for discussion, of course) :
1. have a com.xwiki.admintools.internal.wikiSize package, to include also the InstanceUsage class
2. have a WikiSize class that has a method `public WikiSizeResult getWikiSize(String wikiId)`, which will give the information for one wiki
3. have inside InstanceUsage the `private List<WikiSizeResult> getWikisSizeInfo()` that just calls wikiSize.getWikiSize(wikiId) for each wiki